diff --git a/libs/platform/CMakeLists.txt b/libs/platform/CMakeLists.txt
index c26e864b..10ab3009 100644
--- a/libs/platform/CMakeLists.txt
+++ b/libs/platform/CMakeLists.txt
@@ -72,11 +72,11 @@ if (PLATFORM_IPHONE)
     http_thread_apple.mm
     http_client_apple.mm
     http_uploader_apple.mm
-    http_user_agent_ios.mm
     localization.mm
     locale.mm
     network_policy_ios.h
     network_policy_ios.mm
+    http_session_manager.mm
     platform_ios.mm
     platform_unix_impl.cpp
     platform_unix_impl.hpp
