diff --git a/libs/search/CMakeLists.txt b/libs/search/CMakeLists.txt
index 94da4418..98fa387c 100644
--- a/libs/search/CMakeLists.txt
+++ b/libs/search/CMakeLists.txt
@@ -179,7 +179,7 @@ target_link_libraries(${PROJECT_NAME}
   openlocationcode
 )
 
-if(PLATFORM_DESKTOP)
+if(PLATFORM_DESKTOP AND NOT SKIP_TESTS)
   add_subdirectory(search_tests_support)
   add_subdirectory(search_quality)
 endif()
