diff --git a/libs/search/latlon_match.cpp b/libs/search/latlon_match.cpp
index 6e6bdb8..811291f 100644
--- a/libs/search/latlon_match.cpp
+++ b/libs/search/latlon_match.cpp
@@ -118,7 +118,7 @@ double EatDouble(char const * str, char ** strEnd)
       gotMark = true;
       markPos = p;
     }
-    else if (isdigit(*p))
+    else if (isdigit(static_cast<unsigned char>(*p)))
     {
       if (gotMark)
         gotDigitAfterMark = true;
